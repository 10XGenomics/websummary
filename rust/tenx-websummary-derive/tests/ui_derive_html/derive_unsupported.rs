use tenx_websummary_derive::HtmlTemplate;

#[derive(HtmlTemplate)]
enum Foo {
    Bar,
}

#[derive(HtmlTemplate)]
struct Boo(u8);

fn main() {}
