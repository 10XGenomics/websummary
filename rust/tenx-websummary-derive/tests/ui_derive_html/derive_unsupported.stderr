error: HtmlTemplate can only be derived for structs with named fields. Unsupported shape `enum`. Expected struct with named fields.
 --> tests/ui_derive_html/derive_unsupported.rs:3:10
  |
3 | #[derive(HtmlTemplate)]
  |          ^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `HtmlTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: HtmlTemplate can only be derived for structs with named fields. Unsupported shape `one unnamed field`. Expected named fields.
 --> tests/ui_derive_html/derive_unsupported.rs:8:10
  |
8 | #[derive(HtmlTemplate)]
  |          ^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `HtmlTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)
